force-app/main/default/classes/AccountTriggerHandler.cls

```diff
@@ -0,0 +1,82 @@
+        
+        for(Account acc : (List<Account>)Trigger.New){
+            Opportunity opp = new Opportunity(Name=acc.Name + ' ' + date.today().month() + ' ' + date.today().year(), 
+            StageName='Prospecting', CloseDate=System.today().addDays(90), AccountId=acc.Id, Discount_Percent__c=0.6, Amount = 1);
+            System.debug(acc.Id);
+            oppsToInsert.add(opp);
+        }
+
+        insert oppsToInsert;
+    }
+    public override void afterUpdate() {
+        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+        Map<Id, Account> mapAccs = new Map<Id, Account>();
+
+        for(Account acc : (List<Account>)Trigger.New){
+            if(acc.Name != ((Map<Id, Account>)Trigger.OldMap).get(acc.Id).Name){
+                mapAccs.put(acc.Id, acc);
+            }
+        }
+        
+        List<Opportunity> newOpps = [
+            SELECT Name, AccountId 
+            FROM Opportunity 
+            WHERE AccountId 
+            IN :myMap.keySet()
+        ];
+        
+        for(Opportunity opp : newOpps){
+            if(!opp.Name.contains(mapAccs.get(opp.AccountId).Name)){
+                opp.Name = mapAccs.get(opp.AccountId).Name + ' ' + mapAccs.get(opp.AccountId).CreatedDate.month() + ' ' + mapAccs.get(opp.AccountId).CreatedDate.year();
+                oppsToUpdate.add(opp);
+            }
+        }
+
+        update oppsToUpdate;
+    }
+    public override void beforeDelete() {
+        List<Opportunity> oppsToDelete = [
+            SELECT Name, Amount, AccountId FROM Opportunity 
+            WHERE AccountId 
+            IN :((Map<Id, Account>)Trigger.OldMap).keySet()
+        ];
+        EmailTemplate et = [
+            SELECT id, body, subject
+            FROM EmailTemplate 
+            WHERE Name =: 'Account Deletion Template' 
+            LIMIT 1
+        ];
+
+        Messaging.SingleEmailMessage message = new Messaging.SingleEmailMessage();
+        message.setTemplateId(et.Id);
+        User activeUser = [
+            SELECT Email 
+            FROM User 
+            WHERE UserName =: UserInfo.getUserName() 
+            LIMIT 1
+        ];
+        message.setTargetObjectId(activeUser.Id);
+        message.saveAsActivity = false;
+
+        String htmlBody = et.body;
+        String opps_list = '';
+        Integer i = 0;
+
+        for(Opportunity opp : oppsToDelete){
+            i++;
+            opps_list += i + '. ' + opp.id + ', ' + opp.Name + ', ' + opp.amount + '.\n';
+            
+            htmlBody =  htmlBody.replace('{AccountName}', ((Map<Id, Account>)Trigger.OldMap).get(opp.AccountId).Name);
+            htmlBody = htmlBody.replace('{opps_list}', opps_list);
+            htmlBody = htmlBody.replace('{BestRegards}', Label.best_regards);
+            message.setPlainTextBody(htmlBody);
+            message.setToAddresses(new String[] {activeUser.Email});
+            Messaging.sendEmail(new Messaging.SingleEmailMessage[]{message});
```

Comment:
Email must be sent for opportunities per account. All at once.

Reply:
this way you don't need Integer variable. Try to map opps by Account and make use of list size.

Reply:
this way you don't need Integer variable. Try to map opps by Account and make use of list size.